### PR TITLE
Change filereader

### PR DESCRIPTION
**Pull request by szymon-filipiak**
This merge deals with memory related improvements. Actually it solves the issue of program malfunctioning randomly. Please check @zawadzki-dawid @WiktorPieklik

### CODE REVIEW

**Comment by szymon-filipiak** [comment]
Also I've deleted the templated version of the algorithm class. We can get back to it but for now I think it's just making debugging harder. Please let me know what do you think

**Review by zawadzki-dawid — APPROVED** [review]

File: src/Text/TextFileReader.cpp [hunk 1]
```diff
@@ -1,21 +1,21 @@
...
-    std::string line;
-    while(std::getline(file, line))
-    {
-        textVector.push_back(line);
-    }
-
-    return textVector;
+    std::ifstream file(file_path);
+    if(!file.is_open())
+        throw std::runtime_error("Could not open file: " + file_path.string());
```
Comment (WiktorPieklik) [inline src/Text/TextFileReader.cpp:16]: I'll change `std::runtime_error` to our `FileNotFoundException`

**Review by WiktorPieklik — APPROVED** [review]
Fantastic job! 🥇

File: src/Text/TextFileReader.cpp (hunk 1 above)
Comment (WiktorPieklik) [reply src/Text/TextFileReader.cpp:16]: Sorry, now I understand this statement. Forget my comment 😆

**Pull request merged**